packages/twenty-server/src/modules/workflow/workflow-builder/utils/generate-fake-object-record-event.ts

```diff
@@ -2,27 +2,30 @@ import { v4 } from 'uuid';
 
 import { DatabaseEventAction } from 'src/engine/api/graphql/graphql-query-runner/enums/database-event-action';
 import { ObjectMetadataEntity } from 'src/engine/metadata-modules/object-metadata/object-metadata.entity';
+import {
+  Leaf,
+  Node,
+} from 'src/modules/workflow/workflow-builder/types/output-schema.type';
 import { generateFakeObjectRecord } from 'src/modules/workflow/workflow-builder/utils/generate-fake-object-record';
-import { OutputSchema } from 'src/modules/workflow/workflow-builder/types/output-schema.type';
 
 export const generateFakeObjectRecordEvent = (
   objectMetadataEntity: ObjectMetadataEntity,
   action: DatabaseEventAction,
-): OutputSchema => {
+): Record<string, Leaf | Node> => {
   const recordId = v4();
   const userId = v4();
   const workspaceMemberId = v4();
 
   const after = generateFakeObjectRecord(objectMetadataEntity);
   const formattedObjectMetadataEntity = Object.entries(
     objectMetadataEntity,
-  ).reduce((acc: OutputSchema, [key, value]) => {
+  ).reduce((acc: Record<string, Leaf | Node>, [key, value]) => {
     acc[key] = { isLeaf: true, value };
 
     return acc;
   }, {});
```

Comment:
This might not be the scope of this PR, but I prefer using a for loop than a reduce function, which is harder to read.

This could be:

```tsx
  const formattedObjectMetadataEntity: Record<string, Leaf | Node> = {};

  for (const [key, value] of Object.entries(objectMetadataEntity)) {
    formattedObjectMetadataEntity[key] = { isLeaf: true, value };
  }
```

Reply:
I use reduce a lot actually. This is definitely a discussion we should have if you do not like it. But looks like a global strategy rather than just a local change :)

Reply:
That's definitely a stylistic choice and isn't that important. I'm totally okay if we keep using it. I just want to explain my reasoning behind it.

I have two concerns with reduce: it usually leads to less understandable code – in this case, the reduce code is simple and remains understandable –, and it can lead to bugs when not complying with function programming patterns by returning a new instance every time.

Here, we directly mutate the accumulator instead of returning a new accumulator every time. If we used a for loop, it would be obvious that we actually mutate the JavaScript reference. Accidentally mutating references when using React is an insidious source of bugs and I like to make mutation as obvious as possible.

Taken from the mdn's article on [Array#reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce):

> <code>reduce()</code> is a central concept in <a href="https://en.wikipedia.org/wiki/Functional_programming" class="external" target="_blank">functional programming</a>, where it's not possible to mutate any value, so to accumulate all values in an array, one must return a new accumulator value on every iteration. This convention propagates to JavaScript's <code>reduce()</code>: you should use <a href="/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax">spreading</a> or other copying methods where possible to create new arrays and objects as the accumulator, rather than mutating the existing one. If you decide to mutate the accumulator instead of copying it, remember to return the modified object in the callback still, or the next iteration will receive undefined. However, note that copying the accumulator may in turn lead to increased memory usage and degraded performance — see <a href="#when_to_not_use_reduce">When to not use reduce()</a> for more details. In such cases, to avoid bad performance and unreadable code, it's better to use a <code>for</code> loop instead.